06-descriptive-analysis-in-srvyr.Rmd

```diff
@@ -1 +1,420 @@
+* `name`: the name of the count variable, defaults to `n`
+* `.drop`: whether to drop empty groups
+* `vartype`: type(s) of variation estimate to calculate, defaults to `se` (standard error)
+
+The steps to use `survey_count()` are:
+
+* Specify the sample design,
+* Filter subsets using `dplyr::filter()`, if needed
+* Run `survey_count()`, specifying the required arguments within the function
+
+Let's see the weighted count of responses in ANES:
+
+```{r}
+anes_des %>% #  Specify the sample design
+  survey_count() # Run `survey_count()`
+```
+
+`srvyr::count()` can take one or many variables. To calculate a cross-tab of population in each age group and gender, we run the below:
+
+```{r}
+anes_des %>%
+  # Specify the required arguments within the function
+  survey_count(AgeGroup, Gender, name = "N") 
+```
+
+### Calculate totals using survey methods using `survey_total()`
+
+With `srvyr::survey_total()`, we can calculate totals from complex survey data. Let's explore the syntax:
+
+```r
+survey_total(
+  x,
+  na.rm = FALSE,
+  vartype = c("se", "ci", "var", "cv"),
+  level = 0.95,
+  deff = FALSE,
+  df = NULL,
+  ...
+)
+```
+
+* `x`: a variable, expression, or empty
+* `na.rm`: an indicator of whether missing values should be dropped, defaults to `FALSE`
+* `vartype`: type(s) of variation estimate to calculate, defaults to `se` (standard error)
+* `level`: a number or a vector indicating the confidence level, defaults to 0.95
+* `deff`: a logical value stating whether the design effect should be returned, defaults to FALSE
+* `df`: for `vartype = 'ci'`), a numeric value indicating degrees of freedom for the t-distribution
+  * For the {srvyr} package, this defaults to `NULL` whereas the {survey} package defaults to `Inf`
+
+The steps to use `survey_total()` are:
+
+* Specify the sample design,
+* Specify the cross tab in `group_by()`,
+* Within `summarize`, run `survey_total()`, specifying the required arguments within the function
+
+To calculate a population count estimate with `survey_total()`, we can run the below:
+
+```{r}
+anes_des %>%
+  summarize(survey_total(), .groups = "drop")
+```
+
+The `.groups =` argument controls the grouping structure of the output. When the output no longer have grouping variables because they are dropped, it becomes ungrouped. <!-- TODO: Check this after dplyr 1.1.0 is released -->
+
+Notice that `anes_des %>% summarize(survey_total(), .groups = "drop")` is equivalent to the `survey_count()` call:
+
+```{r}
+anes_des %>%
+  survey_count() 
+```
+
+The `survey_total()` function is called within `summarize`, where as `survey_count()`, like `dplyr::count()`, is not.
+
+### Calculate mean/proportion using survey methods with `survey_mean()` and `survey_prop()`
+
+The `srvyr::survey_mean()` and `survey_prop()` functions calculate the means and proportions from complex survey data. Like `survey_total()`, they are called within `summarize()`. Let's explore the syntax:
```

Comment:
Bring in an example with survey_mean. Need something like RECS with continuous outcomes.